crawler.go

```diff
@@ -122,10 +142,22 @@ func (c *Crawler) EnqueueWithSource(method string, URL string, sourceURL string)
 		if err != nil {
 			return ok, err
 		}
-		cmd := Cmd{&fetchbot.Cmd{U: u, M: "GET"}, s}
-		err = c.q.Send(cmd)
-		return ok, err
+
+		if c.BasicAuthUser != "" && c.BasicAuthPass != "" {
+			cmd := &CmdBasicAuth{&fetchbot.Cmd{U: u, M: method}, s, c.BasicAuthUser, c.BasicAuthPass}
```

Comment:
Please DRY this, just assign `cmd` in `if else` statement, then you can `Send` it.